### PR TITLE
Dereference Object Properly

### PR DESCRIPTION
**Pull request by qur786**

### CODE REVIEW

**Review by maneetgoyal — CHANGES_REQUESTED** [review]
`.map` is better than `.forEach` in some cases because of its conciseness (eliminates the need for `.push()`.

File: src/deref.ts [hunk 1]
```diff
@@ -11,46 +11,53 @@ import {
...
-    elName: input.elname.toArray() as ATOMELLENTuple,
-    numBonds: input.num_bonds,
-    numIsoH: input.num_iso_H.toArray() as NUMHISOTOPESTuple,
-    isotopicMass: input.isotopic_mass,
-    radical: input.radical,
-    charge: input.charge,
-  };
+export function generateINCHIAtom(input: ReturnType<typeof inchi_Atom>[]): INCHIAtom[] {
+  let ouput: INCHIAtom[] = [];
+  input.forEach((element) => {
```
[the remark this answers is not in the document]
Comment (qur786) [reply src/deref.ts:22]: sure sir , @maneetgoyal  sir can you look at the keys of inchi_Input_V3000 mainly lists_haptic_bonds , lists_steabs , lists_sterel , lists_sterac . Those keys are pointer to array pointers and I have implemented them as array of array . Is it correct ?

[the remark this answers is not in the document]
Comment (maneetgoyal) [reply src/deref.ts:22]: @qur786 

- `lists_haptic_bonds`: Should be treated as a pointer to an array.
- `lists_steabs`, `lists_sterel`, `lists_sterac`: Should be treated as array of arrays.

[the remark this answers is not in the document]
Comment (qur786) [reply src/deref.ts:22]: Ok sir

File: src/headers.ts [hunk 2]
```diff
@@ -730,19 +730,19 @@ export const inchi_Input_V3000 = NAPIStructType({
...
-  lists_sterac: refNAPI.refType(refNAPI.refType(refNAPI.types.int)),
+  lists_sterac: NAPIArrayType(NAPIArrayType(refNAPI.types.int)),
 });
 export interface INCHIInputV3000 {
   nNonStartAtoms: number;
   nStarAtoms: number;
-  atomIndexOrig: number;
-  atomIndexFin: number;
+  atomIndexOrig: number[];
+  atomIndexFin: number[];
```
Comment (maneetgoyal) [inline src/headers.ts:745]: Revert this change.

File: src/headers.ts [hunk 3]
```diff
@@ -781,13 +781,13 @@ export interface INCHIInputV3000 {
...
+  Stereo0D: NAPIArrayType(inchi_Stereo0D),
   szOptions: refNAPI.types.CString,
   num_atoms: refNAPI.types.short,
   num_stereo0D: refNAPI.types.short,
-  polymer: refNAPI.refType(inchi_Input_Polymer),
-  v3000: refNAPI.refType(inchi_Input_V3000),
+  polymer: NAPIArrayType(inchi_Input_Polymer),
+  v3000: NAPIArrayType(inchi_Input_V3000),
 });
 export interface INCHIInputEx {
```
Comment (maneetgoyal) [inline src/headers.ts:792]:
```ts
export interface INCHIInputEx {
  atom: INCHIAtom;
  stereo0D: INCHIStereo0D | null;
  szOptions: string;
  numAtoms: number;
  numStereo0D: number;
  polymer: INCHIInputPolymer | null;
  v3000: INCHIInputV3000 | null;
}
```

File: src/headers.ts [hunk 4]
```diff
@@ -926,26 +926,26 @@ export interface INCHIOutputStruct {
...
   szMessage: refNAPI.types.CString,
   szLog: refNAPI.types.CString,
   WarningFlags: NAPIArrayType(NAPIArrayType(refNAPI.types.ulong, 2), 2),
-  polymer: refNAPI.refType(inchi_Input_Polymer),
-  v3000: refNAPI.refType(inchi_Input_V3000),
+  polymer: NAPIArrayType(inchi_Input_Polymer),
+  v3000: NAPIArrayType(inchi_Input_V3000),
 });
 export interface INCHIOutputStructEx {
-  atom: INCHIAtom;
```
Comment (maneetgoyal) [inline src/headers.ts:940]:
```ts
export interface INCHIOutputStruct {
  atom: INCHIAtom[];
  stereo0D: INCHIStereo0D[] | null;
  numAtoms: number;
  numStereo0D: number;
  szMessage: string | null;
  szLog: string | null;
  warningFlags: [[number, number], [number, number]];
}
```

File: src/headers.ts [hunk 5]
```diff
@@ -926,26 +926,26 @@ export interface INCHIOutputStruct {
...
+  atom: INCHIAtom[];
+  stereo0D: INCHIStereo0D[];
   numAtoms: number;
   numStereo0D: number;
   szMessage: string | null;
   szLog: string | null;
   warningFlags: [[number, number], [number, number]];
-  polymer: INCHIInputPolymer;
-  v3000: INCHIInputV3000;
+  polymer: INCHIInputPolymer[];
```
Comment (maneetgoyal) [inline src/headers.ts:947]:
```ts
export interface INCHIOutputStructEx {
  atom: INCHIAtom[];
  stereo0D: INCHIStereo0D[] | null;
  numAtoms: number;
  numStereo0D: number;
  szMessage: string | null;
  szLog: string | null;
  warningFlags: [[number, number], [number, number]];
  polymer: INCHIInputPolymer[];
  v3000: INCHIInputV3000[];
}
```

**Review by maneetgoyal — CHANGES_REQUESTED** [review]
